cloud/pkg/admissioncontroller/common.go

```diff
@@ -64,44 +65,44 @@ func registerMutatingWebhook(client admissionregistrationv1beta1client.MutatingW
 type hookFunc func(admissionv1beta1.AdmissionReview) *admissionv1beta1.AdmissionResponse
 
 func serve(w http.ResponseWriter, r *http.Request, hook hookFunc) {
-	var body []byte
-	if r.Body != nil {
-		if data, err := io.ReadAll(r.Body); err == nil {
-			body = data
-		}
-	}
+	// The AdmissionReview that will be returned
+	var responseAdmissionReview admissionv1beta1.AdmissionReview
 
 	// verify the content type is accurate
 	contentType := r.Header.Get("Content-Type")
 	if contentType != "application/json" {
-		klog.Exitf("contentType=%s, expect application/json", contentType)
-		return
-	}
-
-	// The AdmissionReview that was sent to the webhook
-	requestedAdmissionReview := admissionv1beta1.AdmissionReview{}
-
-	// The AdmissionReview that will be returned
-	responseAdmissionReview := admissionv1beta1.AdmissionReview{}
-
-	deserializer := codecs.UniversalDeserializer()
-	if _, _, err := deserializer.Decode(body, nil, &requestedAdmissionReview); err != nil {
-		klog.Exitf("decode failed with error: %v", err)
+		err := fmt.Errorf("contentType=%s, expect application/json", contentType)
```

Comment:
we can put validate the content type in the first and return err if it is invalid 
sugget 


````
	// validate the content type
	contentType := r.Header.Get("Content-Type")
	if contentType != "application/json" {
		klog.Errorf("contentType=%s, expect application/json", contentType))
		http.Error(w, "invalid Content-Type, want `application/json`", http.StatusUnsupportedMediaType)
		return
	}
```